azure-pipelines.yml

```diff
@@ -0,0 +1,486 @@
+        _PortableBuild: true
+        _TargetArchitecture: arm64
+      pool:
+        ${{ if eq(variables['System.TeamProject'], 'public') }}:
+          name: NetCorePublic-Int-Pool
+          queue: Windows.10.Amd64.Open
+        ${{ if ne(variables['System.TeamProject'], 'public') }}:
+          name: dotnet-internal-temp
+      strategy:
+        maxParallel: 2
+        matrix: 
+          debug:
+            _BuildConfiguration: Debug
+          release:
+            _BuildConfiguration: Release
+
+    # Windows Arm
+    - job:
+      displayName: Windows_Arm
+      agentOS: Windows
+      variables:
+        _AdditionalMSBuildArguments: "/p:SkipTests=true  /p:StabilizePackageVersion=$(IsStable)"
+        _PortableBuild: true
+        _TargetArchitecture: arm
+      pool:
+        ${{ if eq(variables['System.TeamProject'], 'public') }}:
+          name: NetCorePublic-Int-Pool
+          queue: Windows.10.Amd64.Open
+        ${{ if ne(variables['System.TeamProject'], 'public') }}:
+          name: dotnet-internal-temp
+      strategy:
+        maxParallel: 2
+        matrix: 
+          debug:
+            _BuildConfiguration: Debug
+          release:
+            _BuildConfiguration: Release
+
+# Platform: OSX
+- template: /eng/jobs/osx-build.yml
+  parameters:
+    pool:
+      ${{ if eq(variables['System.TeamProject'], 'public') }}:
+        name: Hosted macOS
+      ${{ if ne(variables['System.TeamProject'], 'public') }}:
+        name: Hosted Mac internal
+    strategy:
+      maxParallel: 5
+      matrix: 
+        OSX-x64-debug:
+          _AdditionalBuildArguments: "-PortableBuild=true -strip-symbols -SkipTests=$(SkipTests)"
+          _AdditionalMSBuildArguments: " /p:StabilizePackageVersion=$(IsStable)"
+          _BuildConfiguration: Debug
+          _PortableBuild: true
+        OSX-x64-release:
+          _AdditionalBuildArguments: "-PortableBuild=true -strip-symbols -SkipTests=$(SkipTests)"
+          _AdditionalMSBuildArguments: " /p:StabilizePackageVersion=$(IsStable)"
+          _BuildConfiguration: Release
+          _PortableBuild: true
+          _TargetArchitecture: x64
+
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+  # Signing Validation
+  - template: /eng/jobs/validate-signing.yml
+    parameters:
+      pool:
+        ${{ if eq(variables['System.TeamProject'], 'public') }}:
+          name: NetCorePublic-Int-Pool
+          queue: Windows.10.Amd64.Open
+        ${{ if ne(variables['System.TeamProject'], 'public') }}:
+          name: dotnet-internal-temp
+      dependsOn: 
+          - Linux_x64_glibc
+          - Linux_Arm64
+          - Linux_Arm
+          - Linux_x64_Alpine36
+          # - Linux_x64_FreeBSD
+          - Linux_x64_Rhel6
+          - Windows_x64
+          - Windows_x86
+          - Windows_Arm
+          - Windows_Arm64
+          - OSX
+  # Finalize Publish 
+  - template: /eng/jobs/finalize-publish.yml
```

Comment:
Why are any parameters needed here?  Isn't finalize-publish only invoke in one place?  Can these just be defined within the template?  This comment applies to any template that is only used once.

Reply:
My initial thought was to keep input yml (azure-pipelines.yml) consistent across build legs. This improves readability of the file w.r.t to dependency and pool as it provides overview of job flow for the entire pipeline. Removing parameters and placing the values in place does reduce the overall length of the file. As per suggestion, will update all the jobs that are called only once